### ISSUE TITLE
Provide a (real) self hosted version

### ISSUE THREAD

**Issue by ghost** [sanity-io/sanity#3258]
We would love to see a self-managed version of sanity, so that we can install the sanity system on our own servers (for example via docker). Not just the frontend, but the whole app including backend, database and assets.

The Ys:
- don’t want to store data at servers of big US companies (they are called google and amazon)
- let’s decentralize sanity and our data
- keep data in our countries (europe)
- respect privacy regulations of our countries and not the ones of the US (are there any?)

We really love sanity, because it is maybe the best (80%) admin panel out there.
But there is this one aspect that is really missing — a free, open source, self hosted version!

Upvote this feature!
↓

**Comment by adrm** [comment]
Anyone can point to resources about how to run Sanity frontend without depending on privative Sanity backend? AFAIK it's the only way to do it, so not much use in open-sourcing the frontend if you can't use it by itself. Having said that, thanks for taking the open-source route, and I hope you can go through with it.

**Comment by krumware** [comment]
Not being able to airgap or on-prem the content lake is a dealbreaker for some enterprise folks, despite the soc2. Is it at least possible to fully self-host with an enterprise license?

**Comment by JonathanDoelan** [comment]
> Not being able to airgap or on-prem the content lake is a dealbreaker for some enterprise folks, despite the soc2. Is it at least possible to fully self-host with an enterprise license?

@krumware This topic is about FLOSS. So maybe you create your own issue? I think it is a different story.

**Comment by krumware** [comment]
> > Not being able to airgap or on-prem the content lake is a dealbreaker for some enterprise folks, despite the soc2. Is it at least possible to fully self-host with an enterprise license?
> 
> @krumware This topic is about FLOSS. So maybe you create your own issue? I think it is a different story.

I'm happy to, but I don't think it's necessarily different. If the complete stack was available or open then it would be a non-issue. I'm having trouble discovering if it's possible to self host the content lake, which seems to be a critical part of the stack here and matches up with the original issue content.
I certainly don't want to hijack. Should rephrase my previous comment? My goal was to express that it's a desired feature even for enterprise customers that typically adopt and pay for FLOSS support.

However I can best show support for this feature request, that's my goal. Thanks for opening it!

**Comment by JonathanDoelan** [comment]
@krumware I’m sorry for the misconception, it seemed like you wanted the feature available exclusively for enterprise customers.

**Comment by krumware** [comment]
All good, I'll rephrase a bit to try to clarify. Thanks for the feedback!

**Comment by braco** [comment]
Another reason are the API limits, they're not very generous. If you're pushing a lot of small, frequent updates in a large database, those limits are easy to hit

**Comment by TRTester** [comment]
I would love for this feature to be implemented as well!

**Comment by sharpsteelsoftware** [comment]
Would love this I get tired of writing my own sanity dockerfile

**Comment by JonathanDoelan** [comment]
This issue is the 2nd most popular here: https://github.com/sanity-io/sanity/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Just sayin’ …

**Comment by MaffooBristol** [comment]
I can understand why they wouldn't - it would completely destroy their business model. Hosting their own data lake and then charging for extra features is the only thing that brings money in. Just mere speculation of course.

The main issue for me about not being able to self-host is that I'm worried one day Sanity will just disappear into thin air, remove all their data, and everything will just be _gone_. I'm considering moving more and more sites over to Sanity but it scares me a little bit just having all my stuff existing on someone else's servers with no full 100% trust it will be around forever. I guess that's the same for any cloud service, but it's just one I think about.

I guess it's possible to make a custom service that scrapes and backs up the data. And I'm not sure if you can get Sanity to send you a zip of all your data, such as Facebook and the like do?

**Comment by TomFreudenberg** [comment]
> Would love this I get tired of writing my own sanity dockerfile

@sharpsteelsoftware what does this mean?

Are you hosting the sanity API and backend by yourself?

Propably you can share the docker file(s) and more can have an eye on?

Thanks for clarification

Tom

**Comment by fridaystreet** [comment]
Definitely a +1 here. Would love to be able to use Sainity in our application, but without being able to run on prem for data sensitive clients and outside the US means we cannot meet our obligations in regards to those clients data security and sovereignty. 

It doesn't need to be free and affect their business model, if anything enhance it. Enterprise customers are more than willing to pay for the right solution.

**Comment by Brewd3v** [comment]
Had to use something else because of how the data is stored.

**Comment by vedikaorganics** [comment]
Can someone from sanity team comment here atleast?

**Comment by swapkats** [comment]
Its sad to see sanity team not the least bit interested in replying. If your entire business model hinges on not making datastore public, you have bigger things to worry about. Not everyone will want to self host anyway, but I'm definitely not going to use something with shady management like this. That too for a datastorage, its ironic considering and their actual offering is an content management cms. Definitely not going with this since I value data privacy of my users. I'm going to go with something like strapi.

**Comment by raqibnur** [comment]
Must-need thing, We are constantly switching to other self-hosted CMS options.

**Comment by bayraak** [comment]
sanity team: ignorance is bliss

you can sleep well now sanity, I'm out :) self host always wins

**Comment by Fakerko** [comment]
@bayraak What CMS will you use instead of Sanity? I am looking for alternative to Sanity, too.

**Comment by Notorious-Ali** [comment]
@Fakerko Payloadcms

**Comment by Coderamrin** [comment]
Looks like it's time to try payload thanks guys. 
It would have been great if it was possible to self host sanity.

**Comment by kmelve** [comment]
Hi folks! We appreciate the feedback and the desire to self-host Sanity! Let me take a moment to address some of the questions/reasons raised. We haven't replied here because we haven't done a good job of staying on top of GitHub issues across our many repos, and this just came onto our radar.

tl;dr We will unlikely spend the time it would take to engineer a fully self-hosted version of Sanity. We would rather invest that time in improving the developer and editorial experience of the platform that we built.

Some musings from _my_ perspective, having worked at Sanity for six years:
- The backend is fairly advanced and feature-rich, with everything from real-time APIs for data and user presence to scaling the query engine and a complete asset pipeline. We don't think the time it would take to package this neatly and deploy it anywhere is worth it compared to building new features and strengthening the platform.
- Of course, part of our business model is the Software as a Service model, where we sell you hosting of your content, infrastructure for scaling, etc, as part of the service. This allows us to provide a seamless, scalable, and reliable experience out of the box. We believe this is the best way to deliver value to our customers. I don't think offering a self-hosted version would necessarily undercut that business model, but engineering and maintenance efforts would be costly. It always boils down to trade-offs and priorities!
- Likewise, we have seen little indication that not offering Sanity as a self-hosted package is problematic for our business.
Some businesses require their CMS to be installed on-prem. For these, we are probably not the right solution, and that's OK.
- That being said, we can offer hosting on dedicated cloud infrastructure on certain Enterprise plans for those who require it. In general, you can always reach out to us and talk to us about your or your client's requirements.
- [We are GDPR compliant and transparent about our privacy policies](https://www.sanity.io/legal/privacy) etc. We must take these things seriously to be a vendor used by many European public institutions (Sanity started as a Norwegian company). [We went pretty deep into GDPR](https://www.sanity.io/blog/a-rough-guide-to-running-a-gdpr-compliant-saas-business) back when it launched.
- We try to be specific and upfront about what surfaces/parts of Sanity are open-source (like the Studio, GROQ, Glush, Mendoza, Content Source Maps, Portable Text, etc., and _not_ the _hosted_ Content Lake backend), but we understand that this is sometimes lost in translation. If/when we fail to be clear, that's on us.
- With that being said, it's not uncommon for "open-source" CMSes to have Enterprise-only and proprietary features or certain functionality available only as paid plugins, including some of the abovementioned ones.

I hope this was clarifying and not overly defensive. We want to win by offering the best experience for anyone who is building with content, so we really mean it when we say that we appreciate the feedback!

**Comment by TonyTinas** [comment]
Still the most popular issue here: https://github.com/sanity-io/sanity/issues?q=is%3Aissue%20sort%3Areactions-desc

I respect the decision of the Sanity Team. For me, this means moving to a more powerful CMS. I have worked with Payload and Strapi, but for me the system that solves all my advanced problems is Directus!

**Comment by Fakerko** [comment]
I also respect the statement and decision of the Sanity team. I will therefore migrate all my clients to another CMS. 

One of my clients has a content site and was paying a rocket for data transfer and API CDN requests. By the way, with competing CDNs the images are cached and they don't have that much data transfer at all.

So I'll be moving to a solution that allows me to host images within the CMS directly on my server or at least integrate with other CDNs.
